tests/api/test_query.py

```diff
@@ -86,3 +87,55 @@ def test_query_kwargs():
 
     with pytest.raises(LookupError):
         query_group_by(group_by='magic')
+
+
+testdata = [
+    (('2018-04-05', '2018-04-06'), Range(begin=datetime.datetime(2018, 4, 5, 0, 0, 0, tzinfo=timezone.utc),
+                                         end=datetime.datetime(2018, 4, 6, 23, 59, 59, 999000, tzinfo=timezone.utc))),
+    ('2008', Range(begin=datetime.datetime(2008, 1, 1, 0, 0, 0, tzinfo=timezone.utc),
```

Comment:
A helper function could make all of these tests more readable. It'd be great to be able to scan down them to see all available options.